astropy/io/ascii/ui.py

```diff
@@ -42,6 +42,51 @@
 _GUESS = True
 
 
+def _probably_html(table, maxchars=100000):
+    """
+    Determine if ``table`` probably contains HTML content.  See PR #3693 and issue
+    #3691 for context.
+    """
+    if not isinstance(table, six.string_types):
+        try:
+            # If table is an iterable (list of strings) then take the first
+            # maxchars of these.  Make sure this is something with random
+            # access to exclude a file-like object
+            table[0]
+            table[:1]
+            size = 0
+            for i, line in enumerate(table):
+                size += len(line)
+                if size > maxchars:
+                    break
+            table = os.linesep.join(table[:i+1])
+        except:
+            pass
+
+    if isinstance(table, six.string_types):
+        # Look for signs of an HTML table in the first maxchars characters
+        table = table[:maxchars]
+
+        # URL ending in .htm or .html
+        if re.match(r'http[s]?:// .+ \.htm[l]?$', table, re.IGNORECASE):
```

Comment:
Also, I think should have `^` at beginning so it won't match on bogus protocols like `Xhttp://`.  Should probably also accept `ftp` and `file`.  Or maybe just `[a-z]{1-9}` for the protocol.